### PR TITLE
feat: upgrade to openapi3-ts@0.11.0

### PR DESCRIPTION
**Pull request by raymondfeng**
The release of openapi3-ts@0.11.0 closes the gap of OAS 3.0 spec.
This PR removes most of our own type declarations for the spec.

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/openapi-v3-types/src/openapi-v3-spec-types.ts [hunk 1]
```diff
@@ -45,336 +50,6 @@ export type ParameterStyle =
...
-
-/**
- * Describes a single operation parameter.
- * A unique parameter is defined by a combination of a name and location.
- * Specification
- * https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameter-object
- */
-export interface ParameterObject extends ISpecificationExtension {
-  name: string;
-  in: ParameterLocation;
```
Comment (jannyHou) [inline packages/openapi-v3-types/src/openapi-v3-spec-types.ts]: @raymondfeng The `openapi3-ts` package doesn't have `ParameterLocation` and `ParameterStyle`, so its `ParameterObject` doesn't verify the value of property `in` and `style` are valid

File: packages/openapi-v3-types/src/openapi-v3-spec-types.ts [hunk 2]
```diff
@@ -21,6 +22,10 @@ export type OpenApiSpec = OAS3.OpenAPIObject;
 // tslint:disable-next-line:no-any
 export type ExtensionValue = any;
 
+export type SchemasObject = {
```
Comment (bajtos) [inline packages/openapi-v3-types/src/openapi-v3-spec-types.ts]: I think this should be called `SchemaObject` (singular Schema)? The old code has `export interface SchemaObject`.

File: packages/openapi-v3-types/src/openapi-v3-spec-types.ts [hunk 3]
```diff
@@ -21,6 +22,10 @@ export type OpenApiSpec = OAS3.OpenAPIObject;
 // tslint:disable-next-line:no-any
 export type ExtensionValue = any;
```
Comment (bajtos) [inline packages/openapi-v3-types/src/openapi-v3-spec-types.ts]: AFAICT, `ExtensionValue` type is no longer used, could you please remove it too?

**Comment by raymondfeng** [comment]
See https://github.com/metadevpro/openapi3-ts/pull/20

File: packages/openapi-v3-types/src/openapi-v3-spec-types.ts (hunk 2 above)
Comment (raymondfeng) [reply packages/openapi-v3-types/src/openapi-v3-spec-types.ts]: There are two types: SchemaObject (for schema definitions) and SchemasObject (for `schemas` of the spec).

File: packages/openapi-v3-types/src/openapi-v3-spec-types.ts (hunk 3 above)
Comment (raymondfeng) [reply packages/openapi-v3-types/src/openapi-v3-spec-types.ts]: It's still referenced in our code base to avoid `any`.

**Comment by raymondfeng** [comment]
Please `openapi3-ts@0.11.0` is released with my patches.

**Comment by bajtos** [comment]
👏

File: packages/openapi-v3-types/src/openapi-v3-spec-types.ts (hunk 3 above)
Comment (bajtos) [reply packages/openapi-v3-types/src/openapi-v3-spec-types.ts]: See https://github.com/strongloop/loopback-next/pull/1271

**Pull request merged**